src/parse-url.ts

```diff
@@ -16,32 +18,30 @@
  * }
  */
 export default function parseURL(url: string) {
-  // TODO: something for when document isn't present... #yolo
-  var anchor = document.createElement('a');
-  anchor.href = url;
+  var parsedUrl = new ParsedUrl(url);
 
-  if (!anchor.host) {
+  if (!parsedUrl.host) {
     // eslint-disable-next-line no-self-assign
-    anchor.href = anchor.href; // IE: load the host and protocol
+    parsedUrl.href = parsedUrl.href; // IE: load the host and protocol
   }
 
-  var pathname = anchor.pathname;
+  var pathname = parsedUrl.pathname;
   if (pathname.charAt(0) !== '/') {
     pathname = '/' + pathname; // IE: prepend leading slash
   }
 
-  var host = anchor.host;
-  if (anchor.port === '80' || anchor.port === '443') {
-    host = anchor.hostname; // IE: remove default port
+  var host = parsedUrl.host;
+  if (parsedUrl.port === '80' || parsedUrl.port === '443') {
+    host = parsedUrl.hostname; // IE: remove default port
```

Comment:
Same as my comment above. These seem to be addressing IE issue, we should have test coverage on the mentioned cases when moving to `url-parse`.
Although there's no cross browser setup yet, but it's still worth adding test for this.